### PR TITLE
feat: ecsact codegen plugin for optimization

### PR DESCRIPTION
**Pull request by zaucy**

### CODE REVIEW

File: ecsact/entt/detail/globals.hh [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+		ecsact::entt::assoc_index,
+		std::unordered_map< //
+			ecsact_component_like_id,
+			decltype(&ecsact_system_execution_context_remove)>>>
+	exec_ctx_remove_fns;
+
+/**
+ * ecsact_system_execution_context_has fn pointers
+ *
+ * NOTE: This hass filled in by ecsact_rt_entt_codegen
```
Comment (zaucy) [inline ecsact/entt/detail/globals.hh:152]:
```suggestion
 * NOTE: This has filled in by ecsact_rt_entt_codegen
```

File: ecsact/entt/detail/globals.hh [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+		ecsact::entt::assoc_index,
+		std::unordered_map< //
+			ecsact_component_like_id,
+			decltype(&ecsact_system_execution_context_has)>>>
+	exec_ctx_has_fns;
+
+/**
+ * ecsact_system_execution_context_action fn pointers
+ *
+ * NOTE: This hass filled in by ecsact_rt_entt_codegen
```
Comment (zaucy) [inline ecsact/entt/detail/globals.hh:166]: this note looks like a copy paste

File: ecsact/entt/wrapper/dynamic.hh [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+template<typename C>
+auto context_update(
+	ecsact_system_execution_context*          context,
+	[[maybe_unused]] ecsact_component_like_id component_id,
+	const void*                               in_component_data
+) -> void {
+	using ecsact::entt::component_changed;
+	// TODO(Kelwan): for remove, beforeremove_storage
```
Comment (zaucy) [inline ecsact/entt/wrapper/dynamic.hh:154]: something unresolved or should this comment be removed?

**Review by zaucy — COMMENTED** [review]
just a few small changes. I haven't fully read it yet. I'll need to run it locally and see the codegen.

One thing I noticed is we're not 'preparing' any of the storages on create registry. We need to do that because otherwise EnTT will generate storage on the fly. This could cause an issue if we run anything in parallel or access storage part way through an iteration conditionally.

you can prepare a storage by simply doing `registry.storage<T>();` and assign it to no variable. Simply calling storage "prepares" it. We should do that in `ecsact_create_registry`.

In the old code it was in [`init_registry`](https://github.com/ecsact-dev/ecsact_rt_entt/blob/46e66f05cdd0b8385104527ace91ebeff50697bb/ecsact/entt/detail/registry_info.hh#L29). As you can see it's 'preparing' each storage ahead of time. It gets called when a registry is created and when it's cleared.

File: ecsact/entt/wrapper/dynamic.hh (hunk 3 above)
Comment (Kelwan) [reply ecsact/entt/wrapper/dynamic.hh:154]: Surprised this showed up for you, I got rid of it before requesting review

Comment (zaucy) [reply ecsact/entt/wrapper/dynamic.hh:154]: maybe you didn't push everything? It's still there.

**Pull request merged**